csharp/extractor/Semmle.Extraction/Context.cs

```diff
@@ -35,12 +35,14 @@ public class Context
         // A recursion guard against writing to the trap file whilst writing an id to the trap file.
         private bool writingLabel = false;
 
+        private readonly Queue<IEntity> labelQueue = new();
+
         protected void DefineLabel(IEntity entity)
         {
             if (writingLabel)
             {
                 // Don't define a label whilst writing a label.
-                PopulateLater(() => DefineLabel(entity));
+                labelQueue.Enqueue(entity);
```

Comment:
Is there a guarantee that `labelQueue` gets emptied? It seems possible that there are items left in the queue:
Thread 1 starts defining label on entity A (`writingLabel == true`), Thread 2 enters `if (writingLabel)`, Thread 1 finishes all the work even the `finally` block, and then Thread 2 enqueues the entity.

Reply:
It's all single threaded: Multi-threading is only for extracting different files in parallel.

Reply:
Hmm, I should have known this. 🤦